### PR TITLE
Update Gradle Wrapper to 8.2

### PR DESCRIPTION
**Pull request by breskeby**
Keep Gradle wrapper up to date

### CODE REVIEW

File: build-tools/src/main/java/org/elasticsearch/gradle/plugin/BasePluginBuildPlugin.java [hunk 1]
```diff
@@ -63,6 +63,7 @@ public BasePluginBuildPlugin(ProviderFactory providerFactory) {
 
     @Override
     public void apply(final Project project) {
+        project.getPluginManager().apply(BasePlugin.class);
```
Comment (mark-vieira) [inline build-tools/src/main/java/org/elasticsearch/gradle/plugin/BasePluginBuildPlugin.java:66]: Why is this required? Does the `JavaPlugin` no longer apply the `BasePlugin`?

**Review by mark-vieira — APPROVED** [review]

File: build-tools/src/main/java/org/elasticsearch/gradle/plugin/BasePluginBuildPlugin.java (hunk 1 above)
Comment (breskeby) [reply build-tools/src/main/java/org/elasticsearch/gradle/plugin/BasePluginBuildPlugin.java]: I thought of applying it explicitly here as we require the BasePluginExtension in this class. But seems to cause more confusion than clarification. Removed it
